lib/pbench/agent/results.py

```diff
@@ -144,32 +155,37 @@ def make_md5sum(self, tarball):
                     self.logger.error(
                         "error removing failed tar ball MD5, {}", tarball_md5
                     )
-            sys.exit(1)
+            return 1
```

Comment:
Instead of returning `1` here, consider re-raising the exception from line 144, possibly chained to a new exception which gives better context (e.g., [RuntimeError](https://docs.python.org/3/library/exceptions.html#RuntimeError), if nothing else, with an appropriate message).

Reply:
Sure :)